### PR TITLE
docs: Remove `Drift` feature gate setting

### PR DESCRIPTION
**Pull request by kedoodle**
Fixes #N/A 

**Description**
Marks `Drift` feature as GA in Feature Gates table, removes `Drift` from example usage of `--feature-gates` per https://github.com/aws/karpenter-provider-aws/blob/73d1a1fe4b5a7d442d0640379a2a646e2cfd240d/website/content/en/preview/upgrading/upgrade-guide.md?plain=1#L69-L70

**How was this change tested?**

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by njtran — COMMENTED** [review]
Looks good! Can you make these changes to the `/docs` and `/v1.0` folders too?

**Comment by engedaam** [comment]
@kedoodle Any updates here?

**Comment by kedoodle** [comment]
> @kedoodle Any updates here?

Hey @engedaam, I had added the change requested by @njtran in b37745b. Is there anything further required?

File: website/content/en/docs/reference/settings.md [hunk 1]
```diff
@@ -41,13 +41,14 @@ Karpenter surfaces environment variables and CLI parameters to allow you to conf
...
-|-------------------------|---------|--------|---------|---------|
-| Drift                   | false   | Alpha  | v0.21.x | v0.32.x |
-| Drift                   | true    | Beta   | v0.33.x |         |
-| SpotToSpotConsolidation | false   | Alpha  | v0.34.x |         |
+Karpenter uses [feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-alpha-or-beta-features) You can enable the feature gates through the `--feature-gates` CLI environment variable or the `FEATURE_GATES` environment variable in the Karpenter deployment. For example, you can configure SpotToSpotConsolidation by setting the CLI argument: `--feature-gates SpotToSpotConsolidation=true`.
+
+| Feature                 | Default | Stage | Since   | Until   |
+|-------------------------|---------|-------|---------|---------|
+| Drift                   | false   | Alpha | v0.21.x | v0.32.x |
+| Drift                   | true    | Beta  | v0.33.x | v1.0.0  |
```
Comment (rschalo) [inline website/content/en/docs/reference/settings.md:49]: Can we actually update the `Until` column to be `v0.37.x`? Otherwise LGTM.

Comment (kedoodle) [reply website/content/en/docs/reference/settings.md]: Done @rschalo 183c896

**Comment by jonathan-innis** [comment]
This appears to be addressed by #7132 so I think that we can close this out now.

**Pull request closed without merging**